jenkins/linux_c_option_test.sh

```diff
@@ -64,6 +64,7 @@ declare -a arr=(
     "-Duse_prov_client:BOOL=ON -Dhsm_type_x509:BOOL=OFF -Dhsm_type_riot:BOOL=ON"
     "-Duse_prov_client:BOOL=ON -Dhsm_type_sastoken:BOOL=ON"
     "-Duse_prov_client:BOOL=ON -Dstrict_prototypes:BOOL=ON"
+    "-DcompileOption_C=\"-Wunused-variable -Wmaybe-uninitialized\""
```

Comment:
```suggestion
    "-Duse_prov_client:BOOL=ON -DcompileOption_C=-Wunused-variable"
     "-Duse_prov_client:BOOL=ON -Wmaybe-uninitialized"
```

Reply:
Easiest option might just be to split them up. The parsing of the combined options seems to be tough for CMake within a bash script.